app/main.go

```diff
@@ -270,39 +378,39 @@ func (app *SpacemeshApp) startSpacemesh(cmd *cobra.Command, args []string) {
 
 	// todo : register all protocols
 
-	err = app.P2P.Start()
-
-	if err != nil {
-		log.Error("Error starting p2p services, err: %v", err)
-		panic("Error starting p2p services")
-	}
+	sgn := hare.NewMockSigning() //todo: shouldn't be any mock code here
+	pub, _ := crypto.NewPublicKey(sgn.Verifier().Bytes())
+	bo := oracle.NewBlockOracle(1, int(app.Config.CONSENSUS.NodesPerLayer), pub.String())
+	hareOracle := oracle.NewHareOracle(1, pub.String())
 
-	app.NodeInitCallback <- true
+	app.P2P = swarm
```

Comment:
this happens 3 times. in line 373 (which is probably the correct one since `app.setupTestFeatures` already consumes `app.P2P`.)
again here and again in initServices